plugins/by-name/rest/default.nix

```
@@ -399,4 +402,31 @@ helpers.neovim-plugin.mkNeovimPlugin {
       ]
     ];
   };
+
+  extraOptions = {
+    curlPackage = lib.mkPackageOption pkgs "curl" {
+      nullable = true;
+    };
+
+    enableHttpFiletypeAssociation = lib.mkEnableOption "filetype association of http files";
+  };
+
+  extraConfig = cfg: {
+    assertions = [
+      {
+        assertion = cfg.enable -> config.plugins.treesitter.enable;
```

```suggestion
        assertion = config.plugins.treesitter.enable;
```

As we discussed earlier. I'd rather not have `cfg.enable` checks in `extraConfig` as it'll confuse people into thinking the config is always defined, even when not enabled.